common/cmake/devices.cmake

```diff
@@ -1537,6 +1708,7 @@ function(ADD_FPGA_TARGET)
     )
     add_custom_target(${NAME}_fasm DEPENDS ${OUT_FASM})
 
+    add_output_to_fpga_target(${NAME} FASM ${OUT_FASM})
```

Comment:
This should be `${OUT_LOCAL_REL}/${TOP}.fasm`, and you should use the FASM property of the target, not `OUT_FASM` when building the dependent file location and deps list